File: android/src/main/kotlin/kth/chem/ke_thermal_printer/core/bluetooth/BluetoothConnectionManager.java
package kth.chem.ke_thermal_printer.core.bluetooth;

import android.bluetooth.BluetoothAdapter;
import android.bluetooth.BluetoothDevice;

import com.rt.printerlibrary.bean.BluetoothEdrConfigBean;
import com.rt.printerlibrary.connect.PrinterInterface;
import com.rt.printerlibrary.enumerate.ConnectStateEnum;
import com.rt.printerlibrary.factory.connect.BluetoothFactory;
import com.rt.printerlibrary.factory.connect.PIFactory;

import kth.chem.ke_thermal_printer.core.PrinterManager;


public class BluetoothConnectionManager {

    public ConnectStateEnum connect(String macAddress) throws Exception {

        BluetoothDevice device = BluetoothAdapter.getDefaultAdapter().getRemoteDevice(macAddress);
        BluetoothEdrConfigBean config = new BluetoothEdrConfigBean(device);
        PIFactory factory = new BluetoothFactory();
        PrinterInterface printerInterface = factory.create();
        printerInterface.setConfigObject(config);

        PrinterManager manager = PrinterManager.getInstance();
        manager.getPrinter().setPrinterInterface(printerInterface);
        manager.getPrinter().connect(config);

        return manager.getPrinter().getConnectState();
    }

    public void disconnect() {
        PrinterManager.getInstance().disconnect();
    }
}

File: android/src/main/kotlin/kth/chem/ke_thermal_printer/core/data/BluetoothConnectionManager.kt
package kth.chem.ke_thermal_printer.core.data

import android.bluetooth.BluetoothAdapter
import android.os.Build
import androidx.annotation.RequiresApi
import com.rt.printerlibrary.bean.BluetoothEdrConfigBean
import com.rt.printerlibrary.connect.PrinterInterface
import com.rt.printerlibrary.factory.connect.BluetoothFactory
import com.rt.printerlibrary.factory.connect.PIFactory
import kth.chem.ke_thermal_printer.core.domain.ConnectionManager

class BluetoothConnectionManager : ConnectionManager {

    @RequiresApi(Build.VERSION_CODES.ECLAIR)
    override fun getConfig(address: String): Any {
        val device = BluetoothAdapter.getDefaultAdapter().getRemoteDevice(address)
        return BluetoothEdrConfigBean(device)
    }

    override fun getInterface(): PrinterInterface<*> {
        val factory: PIFactory = BluetoothFactory()
        return factory.create()
    }
}

